### PR TITLE
DM-39203: Implement a moments-based classifier plugin

### PR DESCRIPTION
**Pull request by arunkannawadi**

### CODE REVIEW

**Review by erykoff — COMMENTED** [review]
A couple docstring comments (easy) and a bigger question about the test thresholds.

File: python/lsst/meas/base/plugins.py [hunk 1]
```diff
@@ -647,10 +641,79 @@ def fail(self, measRecord, error=None):
...
+        object_t = ixx + iyy
+        psf_t = ixx_psf + iyy_psf
+
+        chi_sq = ((object_t - psf_t)/(psf_t)**self.config.exponent)**2.
+        likelihood = 1. - np.exp(-0.5*chi_sq)
+        measRecord.set(self.key, likelihood)
+
+    def fail(self, measRecord, error=None) -> None:
+        # Docstring inherited.
+        measRecord.set(self.key, 1.0)
```
[the remark this answers is not in the document]
Comment (arunkannawadi) [reply python/lsst/meas/base/plugins.py:712]: Yeah, this should be `nan`.

File: python/lsst/meas/base/plugins.py [hunk 2]
```diff
@@ -647,10 +641,79 @@ def fail(self, measRecord, error=None):
...
+
+    ConfigClass = SingleFrameMomentsClassifierConfig
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.FLUX_ORDER
+
+    def __init__(self, config, name, schema, metadata):
+        SingleFramePlugin.__init__(self, config, name, schema, metadata)
+        self.key = schema.addField(name, type="D", doc="Measure of being a galaxy based on trace of second "
```
[the remark this answers is not in the document]
Comment (arunkannawadi) [reply python/lsst/meas/base/plugins.py]: It _is_ trace of the matrix of second moments. This isn't the same as (but related to) `traceRadius` .

File: tests/test_MomentsClassifier.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        star_threshold = {
+            0.001: 0.01,
+            0.01: 0.51
+        }
+        galaxy_threshold = {
+            0.001: 0.25,
+            0.01: 0.25,
+        }
+        self.assertLess(np.mean(star_measures), star_threshold[noise])
+        self.assertLess(np.percentile(star_measures, 50), star_threshold[noise])
```
[the remark this answers is not in the document]
Comment (arunkannawadi) [reply tests/test_MomentsClassifier.py:97]: Ah, I think I had a high value to see what a reasomable value would be. I have changed it to 0.1 to the stars.

**Review by erykoff — APPROVED** [review]

**Pull request merged**